lib/HLSL/DxilValidation.cpp

```diff
@@ -2619,15 +2622,53 @@ static void ValidateGlobalVariable(GlobalVariable &GV,
   }
 }
 
+static void
+CollectFixAddressAccess(Value *V, std::vector<Instruction *> &fixAddrTGSMList) {
+  for (User *U : V->users()) {
+    if (GEPOperator *GEP = dyn_cast<GEPOperator>(U)) {
+      if (isa<ConstantExpr>(GEP) || GEP->hasAllConstantIndices()) {
+        CollectFixAddressAccess(GEP, fixAddrTGSMList);
+      }
+    }
+    if (isa<StoreInst>(U))
+      fixAddrTGSMList.emplace_back(cast<Instruction>(U));
+  }
+}
+
+static void
+ValidateTGSMRaceCondition(std::vector<Instruction *> &fixAddrTGSMList,
+                          ValidationContext &ValCtx) {
+  for (auto &F : ValCtx.DxilMod.GetModule()->functions()) {
```

Comment:
This is computing post-dom for all functions even those that may not have any instructions in the list. It seems like it would be better to only compute dom tree for those functions that have instructions we need to check.